### ISSUE TITLE
Golang: unmarshal error invoking Lambda from Kinesis event record

### ISSUE THREAD

**Issue by mnicolae** [localstack/localstack#3083]
# Type of request: This is a ...

- [x] bug 
- [ ] feature request

# Detailed description

Provisioned localstack environment with Lambda with Kinesis event source trigger. I am using Golang AWS SDK for the integration testing.

```go
// +build component

package component_test

import (
	"fmt"
	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/kinesis"
	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"
	"github.com/sirupsen/logrus"
)

var (
	iLog = logrus.New()

	cfg = aws.NewConfig().WithEndpoint("http://localhost:4566").WithRegion("us-east-1")

	sess *session.Session

	partitionKey = "1"
	streamName = "stream-name"
	kinesisClient *kinesis.Kinesis
)

var _ = Describe("various integration test cases", func() {
	BeforeEach(func() {
		var err error
		sess, err = session.NewSession(cfg)
		if err != nil {
			Fail(fmt.Sprintf("could not init session: %s", err))
		}
		kinesisClient = kinesis.New(sess)
		_, err = kinesisClient.DescribeStream(&kinesis.DescribeStreamInput{
			StreamName: &streamName,
		})
		if err != nil {
			Fail(fmt.Sprintf("could not describe stream: %s", err))
		}
	})

	Context("Test putting valid Kinesis stream records", func() {
		Context("Should trigger associated step function", func() {
			It("dummy unit test", func() {
				putRecordInput := kinesis.PutRecordInput{
					Data: []byte(`{
  "metaData": {
    "eventAt": "2020-10-07T16:49:38.936813404Z",
    "correlationId": "bfa2df06-f22e-4534-8db1-139b10e78355"
  }
}`),
					PartitionKey: &partitionKey,
					StreamName: &streamName,
				}
				iLog.WithField("putRecordInput", putRecordInput).Infof("Logging putRecordInput")
				putRecordOut, err := kinesisClient.PutRecord(&putRecordInput)
				iLog.WithField("putRecordOutput", putRecordOut).Infof("Logging putRecordOutput")
				Expect(err).To(BeNil())
			})
		})


	})
})
```

The lambda is invoked but it fails to unmarshal the `events.KinesisEvent` type.

```
(base) ➜  trigger-claim-lambda git:(IntegrationTesting) ✗ docker logs 8795cef905ad
loading config...
lambdaConf: {0xc000116a80 {trigger-claim-lambda true {true [DEBUG INFO WARN ERROR]}} 0xc000122248}{"level":"debug","msg":"SetupAWSCloud","time":"2020-10-08T03:21:07Z"}
{"level":"debug","msg":"AWS Region: us-east-1","time":"2020-10-08T03:21:07Z"}
{"level":"debug","msg":"AWS SQS Client Connected","time":"2020-10-08T03:21:07Z"}
START RequestId: 5a97d2ac-7cbf-1414-3df4-71111a2bb002 Version: $LATEST
END RequestId: 5a97d2ac-7cbf-1414-3df4-71111a2bb002
REPORT RequestId: 5a97d2ac-7cbf-1414-3df4-71111a2bb002	Init Duration: 55.26 ms	Duration: 5.13 ms	Billed Duration: 100 ms	Memory Size: 1536 MB	Max Memory Used: 25 MB

{"errorType":"UnmarshalTypeError","errorMessage":"json: cannot unmarshal string into Go struct field KinesisRecord.approximateArrivalTimestamp of type float64"}
```

This is likely due to type of `approximateArrivalTimestamp` is string here: https://github.com/localstack/localstack/blob/master/localstack/services/kinesis/kinesis_listener.py#L89

Would be great to know if there is a workaround. Not sure if this reproducible using the CLI or SDKs for other runtimes.

...

## Expected behavior

Lambda handler function should successfully unmarshal the input.

...

## Actual behavior

Lambda handler function returns unmarshal error; due to `approximateArrivalTimestamp` field is string type instead of float64 type.

...

# Steps to reproduce

See Go unit test file above.

## Command used to start LocalStack

```
docker build -t ls compose/localstack
docker-compose up -d
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

```
PROJECT_NAME="trigger-claim-lambda"
STAGE="component"
AWS_LAMBDA_S3_BUCKET="gd-auctions-component-lambdas"
REGION="us-east-1"

# Dependencies
awslocal kinesis create-stream --stream-name "stream-name" --shard-count 1
awslocal s3 mb s3://gd-auctions-${STAGE}-lambdas
awslocal s3 cp ${PROJECT_NAME}.zip s3://${AWS_LAMBDA_S3_BUCKET}
awslocal lambda create-function --function-name gd-dbs-${ENVIRONMENT}-${PROJECT_NAME} --runtime go1.x --role role \
    --handler main --code S3Bucket=${AWS_LAMBDA_S3_BUCKET},S3Key=${PROJECT_NAME}.zip --memory-size 128 --timeout 60 \
    --environment Variables={DEPLOY_ENV=component}
awslocal lambda create-event-source-mapping \
    --event-source-arn arn:aws:kinesis:${REGION}:000000000000:stream/stream-name \
    --function-name gd-dbs-${ENVIRONMENT}-${PROJECT_NAME} --enabled --batch-size 1 --starting-position LATEST
```

**Comment by mnicolae** [comment]
Let me know if there's any other detail I should provide to help with triaging or repro'ing. Thanks.

**Comment by whummer** [comment]
Thanks for reporting @mnicolae . This should be fixed in #3090 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks
